lib/simulator.ml

```diff
@@ -77,17 +78,17 @@ struct
       )
 
   (* Gillespie "direct" method *)
-  let site_gillespie_direct tree ~(root : A.t) ~param =
+  let site_gillespie_direct rng tree ~(root : A.t) ~param =
     let codon_rates = memo (fun b -> M.rate_matrix (param b)) in
     Tree.propagate tree ~init:root ~node:Fn.const ~leaf:Fn.const ~branch:(fun n b ->
         let rec loop state remaining_time =
           let rate_matrix = codon_rates b in
           let rates = A.Table.init (fun m -> if A.equal m state then 0. else rate_matrix.A.%{state, m}) in
-          let total_rate = Owl.Stats.sum (rates :> float array) in
-          let tau = Owl.Stats.exponential_rvs ~lambda:total_rate in
+          let total_rate = Array.fold ~init:0. ~f:(+.) (rates :> float array) in
```

Comment:
This change is not necessary, and it's a performance regression. Here's the bench:
```
┌───────────────┬────────────┬───────────┬──────────┐
│ Name          │   Time/Run │   mWd/Run │  mGC/Run │
├───────────────┼────────────┼───────────┼──────────┤
│ owl_stats_sum │   244.84ns │     2.00w │          │
│ array_fold    │ 2_578.40ns │ 4_000.00w │ 15.26e-3 │
└───────────────┴────────────┴───────────┴──────────┘
```